rest/src/main/kotlin/builder/channel/PermissionOverwritesBuilder.kt

```diff
@@ -0,0 +1,75 @@
+package dev.kord.rest.builder.channel
+
+import dev.kord.common.entity.Overwrite
+import dev.kord.common.entity.OverwriteType
+import dev.kord.common.entity.Snowflake
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+/**
+ * Builder that can modify [permission overwrites][Overwrite].
+ */
+interface PermissionOverwritesBuilder {
+    /**
+     * Adds [overwrite] to this builder.
+     */
+    fun addOverwrite(overwrite: Overwrite)
+}
```

Comment:
Not that we should expect people to implement this, or accept it as a parameter, but this should probably be a sealed interface to prevent future issues where we somehow need to figure out the type of builder ourselves.

```suggestion
sealed interface PermissionOverwritesBuilder {
    /**
     * Adds [overwrite] to this builder.
     */
    fun addOverwrite(overwrite: Overwrite)
}
```

Reply:
Shouldn't Modify and Create be sealed too then?

Reply:
I can almost guarantee that's going to cause issue with our rest<->core modules if we do